### DISCUSSION TITLE
Transition to scikit-build for RapidFuzz

### DISCUSSION THREAD

**Discussion by maxbachmann** [scikit-build/scikit-build#602] [Q&A] [answered]
I am currently transitioning the build of [RapidFuzz](https://github.com/maxbachmann/RapidFuzz) from setuptools to scikit-build, since:
1) all my dependencies use CMake
2) some of the upcoming features like SIMD support would be a pain to get working with setuptools

I added a simple PR implementing most of this [here](https://github.com/maxbachmann/RapidFuzz/pull/153). However it fails on some platforms when building with cibuildwheel and I am unsure how to fix this:
- Windows / PyPy3.7 fails with:
```
-- Found PythonInterp: C:/cibw/pypy3.7-v7.3.5-win64/python.exe (found version "3.7.10")
    CMake Error at C:/Users/runneradmin/AppData/Local/Temp/pip-build-env-xzonggdo/overlay/site-packages/cmake/data/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
      Could NOT find PythonLibs (missing: PYTHON_LIBRARIES) (found version
      "3.7.10")
```
- MacOs PyPy2.7
```
-- Found PythonInterp: /tmp/cibw_bin/python (found version "2.7.18")
    CMake Error at /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/pip-build-env-5lacfO/overlay/site-packages/cmake/data/CMake.app/Contents/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
      Could NOT find PythonLibs (missing: PYTHON_LIBRARIES) (found version
      "2.7.18")
```
- Windows cpython 2.7
```
     File "c:\cibw\python\python2.2.7.18\tools\lib\site-packages\pip\_internal\commands\wheel.py", line 195, in run
    distutils.errors.DistutilsPlatformError: Microsoft Visual C++ 9.0 is required. Get it from http://aka.ms/vcpython27
```
The rest of the builds work fine: https://github.com/maxbachmann/RapidFuzz/actions/runs/1432281305

Any help would be appreciated :smile:

**Comment by maxbachmann** [comment, marked as answer]
I dropped Python2.7, so now only PyPy Windows is still failing to build. However the user base of the library on PyPy/Windows should be close to zero, so this does not really matter. Other than this scikit-build works like a charm and will be used in the next release :partying_face:

**Reply by henryiii** [reply]
I am now finally watching the repo 😳 Glad it's working! FYI, some point last year MS deleted the download for MSVC 9 (which was there for Python only), so it's been impossible to build 2.7 extensions for Windows for many months. Apple removed Python 2 just a few weeks ago, though.

**Reply by maxbachmann** [reply]
Yes I mostly kept support for Python2.7, since it required close to no additional work. As far as I am aware the project did not have many users on Python2.7.
